src/iter.rs

```diff
@@ -226,17 +239,29 @@ impl ExactSizeIterator for HexDigitsIter<'_> {}
 
 impl core::iter::FusedIterator for HexDigitsIter<'_> {}
 
-/// `hi` and `lo` are bytes representing hex characters.
+/// `hi` and `lo` are bytes that encode hex digits as ASCII.
 ///
-/// Returns the valid byte or the invalid input byte and a bool indicating error for `hi` or `lo`.
-fn hex_chars_to_byte(hi: u8, lo: u8) -> Result<u8, (u8, bool)> {
-    let hih = (hi as char).to_digit(16).ok_or((hi, true))?;
-    let loh = (lo as char).to_digit(16).ok_or((lo, false))?;
+/// Returns the valid byte or a bool indicating error for `hi` or `lo`.
+fn hex_digits_to_byte(hi: u8, lo: u8) -> Result<u8, bool> {
+    let hih = (hi as char).to_digit(16).ok_or(true)?;
+    let loh = (lo as char).to_digit(16).ok_or(false)?;
 
     let ret = (hih << 4) + loh;
+
     Ok(ret as u8)
 }
 
+/// The inverse of `hex_digits_to_byte`.
+pub(crate) fn byte_to_hex_digits(byte: u8) -> (u8, u8) {
+    let hi = byte >> 4;
+    let lo = byte & 0x0f;
+
+    let hi = char::from_digit(hi.into(), 16).unwrap();
```

Comment:
In ce28e9ba01cda818439b36108e072435715552fc:

This function is so badly named that I consider it a bug in stdlib. We should just do this manually by indexing into a static array containing `"0123456789abcdef"`. (I am not interested in filing bugs against stdlib to be told that "this is not Haskell" and I shouldn't complain that about words being used wrong. Let's just forget this function and move on with our lives.)

This would also have clearer semantics regarding capitalization, and to the extent there is any performance difference at all, likely to be faster.